### ISSUE TITLE
Alarm Notifications on Attacks

### ISSUE THREAD

**Issue by SecUpwN** [SecUpwN/Android-IMSI-Catcher-Detector#15]
Aside the visual notification of the current ciphering status, I would like AIMSICD to feature a submenu in the settings for enabling a (custom) vibrational pattern or a sound file to be chosen by the user that fires off once an attack of an IMSI-Catcher takes place. A standard subtle vibration pattern shall be included.

---

**UPDATE - Here is the summary of what this Issue shall accomplish:**

1. Add a new menu entry in `Preferences/PROTECTION SETTINGS` called `Detection Alarms`
2. For the [alarm events](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Status-Icons)) `MEDIUM`, `HIGH`, `DANGEROUS` and `RUN!` users shall be able to both:
  * Select a custom ringtone from the phone
  * Be able to set a custom vibrational pattern

To clarify: The user navigates to `Preferences`, hits the button `Detection Alarams` under `PROTECTION SETTINGS` and finds 4 buttons (no alarm on `IDLE` and `NORMAL`) to set the above alarms. Furthermore, we would appreciate a very short vibration for newly updated/connected cells.

:warning: We would like to avoid including tones or any sound files with our app, they just makes it larger.

**Comment by SecUpwN** [comment]
Folks, please have a look at the OP and tell me if you agree with that or have further suggestions.

**Comment by agilob** [comment]
>For each alarm event (each Status Icon) users shall be able to do both of these things: 

There's no point in playing alarm for green and yellow icons. Green is standard, I got yellow basically everyday.

**Comment by E3V3A** [comment]
I agree, and putting a very short vibration for newly updated/connected cells should be trivial.

**Comment by SecUpwN** [comment]
Taken from https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/593#commitcomment-12484913:

>Unfortunately still lacks the option audible indication when capturing an alarm.
From the orange icons or higher...

@AntiDisel, instead of bashing into our proud release, how about crafting a pull request to solve this?

**Comment by E3V3A** [comment]
I've already partially implemented vibration for new events, but it need expansion. Then the concept code for a `NOTIFICATION` type sound would be:
```java
try {
	Uri notification = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
	Ringtone r = RingtoneManager.getRingtone(getApplicationContext(), notification);
	r.play();
} catch (Exception e) {
	e.printStackTrace();
}
```
However, we may prefer to have some kind of very short and subtle *bleep* or *pling* instead, which would require **MediaPlayer** instead, and using a pre-made mp3 file.

```java
Uri notification = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
MediaPlayer mp = MediaPlayer.create(getApplicationContext(), notification);
mp.start();
```
This is better than using **Ringtone**, and prevents interrupting other sounds or music that might already be playing in the background.

**Comment by SecUpwN** [comment]
>However, we may prefer to have some kind of very short and subtle *bleep* or *pling* instead, which would require **MediaPlayer** instead, and using a pre-made mp3 file.

Just to note it here: Please do **NOT** add a pre-made MP3 file into our app, it just enlarges it. I prefer to **generate** the *bleep* or *pling* and think having a look into [IMCKTG](https://github.com/achromaticmetaphor/IMCKTG) may be a good start.

**Comment by E3V3A** [comment]
Well, the default notifications are mp3, no?

**Comment by SecUpwN** [comment]
>Well, the default notifications are mp3, no?

Yes. But I'd like to avoid shipping any MP3s with our app, rather add code to generate a tone by tapping the screen or enable the user to select a tone from his device to avoid bloating our app.

**Comment by SecUpwN** [comment]
Re-opening this Issue since I consider it an important core feature. Maybe @smarek can have a look?

**Comment by vanitasvitae** [comment]
I'd like a really annoying alert when the app detects potential danger. I imagine something comparable to an alarm clock or an incoming phone call. Definitely some persistent sound+vibration that the user has to actively confirm, not just a single "beep" or short vibration. Of course, the user should be able to adjust that option (eg. only activate alert at higher risk levels etc.).

I'm very interested in this project and just forked your code to dig deeper into it.

**Comment by vanitasvitae** [comment]
I did some research and it seems like what I have in mind has to do with a KeyguardManager.KeyguardLock and the PowerManager flag ACQUIRE_CAUSES_WAKEUP.
I will investigate and see, if I can maybe create a class, that creates such an alert.

**Comment by vanitasvitae** [comment]
I did a small mockup of how I'd imagine the alert:
![alert](https://cloud.githubusercontent.com/assets/7854206/12872016/d95fb220-cd94-11e5-9e71-39ce62f6633c.jpg)
The Arrow thingy at the bottom is kind of a slider that the user has to pull aside to stop the alert.

**Comment by SecUpwN** [comment]
>I'm very interested in this project and just forked your code to dig deeper into it.

Welcome to our project and thanks for forking us, @vanitasvitae!

>I'd like a really annoying alert when the app detects potential danger.

Since @larsgrefer has modifies our notifications before to make them *less obtrusive*, I'd like to hear his opinion on this. Generally speaking, I like your idea. But maybe making obtrusiveness an *option* is better?

**Comment by vanitasvitae** [comment]
Thank you @SecUpwN .
Indeed it would be cool to have a checkbox or something in the settings of the app to change notification modes. 

I wanted to say that it has been quite some time since I actively did something for Android and I have to do quite some reading before I can contribute code. Also I am very busy right now with exams, so don't expect some outcomes in the very near future :)

**Comment by SecUpwN** [comment]
>I wanted to say that it has been quite some time since I actively did something for Android and I have to do quite some reading before I can contribute code.

Don't worry. If you see how slow we're currently moving, literally *any* contribution is a small life saver.

>Also I am very busy right now with exams, so don't expect some outcomes in the very near future :)

Take your time and con tribute whenever you feel like your coffee is making you enthusiastic enough. ;-)
